### PR TITLE
Add machine time calculation functions for FPGA

### PR DESCRIPTION
**Pull request by KatieO4t1qbit**
This commit adds an option to calculate the machine_time using fpga. This option can only be used if the solver is LangevinSolver.

Closes https://github.com/1QB-Information-Technologies/ccvm/issues/158

Future commits (other PRs) will add the calculations for other optics solvers.

Testing
You can verify this feature by:

* running the new unit tests
* customizing the solver type and/or the machine parameter of machine_time() in the examples/ccvm_boxqp_plot.py file and then running it

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used, if not `main`

### CODE REVIEW

**Review by ChanWooY-1QBit — COMMENTED** [review]
Just one small minor comment.

File: ccvm_simulators/solvers/langevin_solver.py [hunk 1]
```diff
@@ -284,6 +284,69 @@ def _fpga_machine_energy_callable(matching_df: DataFrame, problem_size: int):
...
+
+            Returns:
+                float: The average time spent by the solver on a single instance.
+            """
+            try:
+                postprocessing_time = np.mean(dataframe["pp_time"].values)
+            except KeyError as e:
+                missing_column = e.args[0]
+                raise ValueError(
+                    f"The given dataframe is missing the {missing_column} column. Required columns are: ['pp_time']."
```
Comment (Estella-Wang-1qbit) [inline ccvm_simulators/solvers/langevin_solver.py:331]: I'm not entirely clear on this error message. If the required column is `pp_time`, wouldn't the missing column in this case be `pp_time`?"

**Review by Estella-Wang-1qbit — APPROVED** [review]
I just have one minor issue with the error message. Other than that, the code looks good to me. Once the comment is resolved, I won't need to review the code again.

File: ccvm_simulators/solvers/langevin_solver.py [hunk 2]
```diff
@@ -284,6 +284,63 @@ def _fpga_machine_energy_callable(matching_df: DataFrame, problem_size: int):
...
+            Raises:
+                ValueError: when the given dataframe does not contain the required
+                    columns.
+
+            Returns:
+                float: The average time spent by the solver on a single instance.
+            """
+            if "pp_time" not in dataframe:
+                raise ValueError(
+                    f"The given dataframe is missing the following columns: pp_time"
```
[the remark this answers is not in the document]
Comment (KatieO4t1qbit) [reply ccvm_simulators/solvers/langevin_solver.py]: Thanks! Fixed.

File: ccvm_simulators/solvers/langevin_solver.py (hunk 1 above)
Comment (KatieO4t1qbit) [reply ccvm_simulators/solvers/langevin_solver.py:331]: That's a great point. I was going for consistency with the other solvers, but for the user that doesn't make a lot of sense. I'll update it!

**Review by FarhadK-1QBit — APPROVED** [review]
LGTM. Thanks, Katie!

**Pull request merged**